### PR TITLE
feat(claude-config): body-scoped coercive-emphasis detector wired to the findings relay

### LINKED ISSUE #3120: feat(claude-config): coercive-emphasis detector on audit-instructions, wired to the findings relay
## Parent

Refs #3118

## What to build

The tracer bullet for the whole container: one narrow, high-confidence detector carried end-to-end through every layer, including the producer wiring that later detectors will reuse.

**The detector (D2, coercive emphasis).** Flag instruction text that steers by volume rather than by content: `CRITICAL:`, `IMPORTANT:` used as a shout, `You MUST`, all-caps imperative words, whole bolded sentences, and blanket over-prompting of the "if in doubt, use X" form. Official re-baselining guidance names this scope directly — dial back aggressive/coercive language, `"CRITICAL: You MUST..."` → `"Use this tool when..."` — and it is the most mechanically detectable of the container's three classes.

**The remediation is a downgrade, never a deletion.** The directive survives verbatim; only its volume changes. A finding that removes the instruction rather than its shouting is wrong.

**The wiring.** `claude-config:audit-instructions` does not currently emit a findings file — today's producers are `ai-slop`, `overengineering`, `review`, `testing`. This slice adds that, per `docs/conventions/detector-findings/README.md`, conforming to the shape owned by `plugins/review/skills/fanout/context/default-mode.md` "Findings-file shape". `ai-slop` is the closest precedent to model on.

Producer obligations the binding leaves to the caller — all four are load-bearing and fail silently when skipped:

- run the full topic-docs rung order, not only its last rung;
- take the non-interactive collapse (a headless detector cannot ask or persist config);
- `branch:` frontmatter proves ownership, never the directory;
- the self-ignore guard is owed, not re-derived, including its invalid cases.

**Body-scoped only.** `plugins/skill-quality/scripts/check-skill.sh:414` hard-FAILs a dropped `'trigger phrase'` versus the base ref. No finding from this detector may have a remediation that edits a `description`, `when_to_use`, or any quoted trigger phrase. A coercive phrase inside a description is reported to the human, never routed to the relay.

**Protected content — never flagged:** directives, thresholds, numeric bounds, qualifiers, version pins, proper nouns, paths, command strings, env vars, motivating context, completion criteria, worked examples.

## Acceptance criteria

- [ ] The coercive-emphasis shape exists in `audit-instructions` with a tier assignment arguable from `review`'s `severity.md` test.
- [ ] The detector writes a conforming `type: review-findings` file; all four producer obligations are implemented, not assumed.
- [ ] End-to-end proof on a test branch: the detector runs, writes findings, `review:fanout fix` locates and applies them. Verified by running it, not by inspection.
- [ ] A negative test proves no finding's remediation touches a description, `when_to_use`, or quoted trigger phrase.
- [ ] A downgrade test proves the directive text survives the remediation byte-for-byte while the coercive wrapper is removed.
- [ ] One no-flag test per protected-content category.
- [ ] `skill-quality:check claude-config:audit-instructions` passes, run against the source checkout (a plugin-cache run applies markdownlint defaults and spuriously fails MD013/MD041/MD060).

## Blocked by

None — can start immediately.

### PR DESCRIPTION
**Pull request by kyle-sexton**
Closes #3120

## Summary

Carries D2 (coercive emphasis) end-to-end to `review:fanout`'s apply relay, plus the findings-producer wiring the `claude-config` plugin did not have. `claude-config:audit-instructions` becomes the plugin's first conforming `detector-findings` producer.

**Scope note worth reading first:** most of the *detection* already existed. `criteria.md` I28 is the coercive-emphasis check, and `instruction-scan.sh` already marked its two families. What was missing was a route to a remediation surface — and a fence making that route safe. The fence turned out to be the substantive part: the scanner read whole files including YAML frontmatter, so `IMPORTANT:` inside a `description:` was a live candidate, directly against #3118's hard constraint.

## Fix

**Body-scope fence, applied twice.** `instruction-scan.sh --body-only` skips YAML frontmatter, so no row can point at a `description`, `when_to_use`, or a trigger phrase quoted in one. It is opt-in, not default: the human-facing audit legitimately reports on frontmatter content, and what must never happen is such a row reaching an *apply* relay. `emit-findings.sh` then **recomputes the fence rather than trusting its input** — a fence living only in the caller is one caller away from being bypassed — and additionally declines any body row quoting a trigger phrase present in the file's own description. An unclosed leading `---` fences the whole file (fail-safe); a mid-document `---` opens nothing. Every decline is counted in `## Surfaces`, never silent.

All three delimiter checks match `^---[[:space:]]*$` — deliberately identical to `skill_frontmatter::extract`, which is what `check-skill.sh` actually parses frontmatter with. Exact `"---"` equality would be *stricter* than that parser, and the mismatch runs the dangerous way (see Review rounds below).

The fence exists because `plugins/skill-quality/scripts/check-skill.sh:414` hard-FAILs a dropped `'trigger phrase'` versus the base ref: a remediation editing one is an auto-invocation regression, not a debatable suggestion.

**Out-of-repo fence.** Phase A inventories user-level surfaces under `${CLAUDE_CONFIG_DIR:-~/.claude}`, but `Location` is contractually repo-relative and the fix action fences each remediation to it. Rows outside the repo root are declined as `reason=outside-repo-root` and stay in the human report.

**Two crosswalk rules**, `rule-coercive-emphasis` and `rule-blanket-tool-default`, both `IMPORTANT`. Argued from `severity.md`'s tests, first match winning: CRITICAL fails every limb because instruction prose computes nothing — what emphasis changes is how likely a trigger is to fire, a distribution over behavior rather than a determined result. IMPORTANT's degradation-with-a-named-trigger limb then matches, the trigger being the first session in which the over-emphasized instruction fires on input its actual condition does not cover. Official guidance supplies the mechanism rather than a preference, so SUGGESTION's catch-all is never reached: the alternatives do not both work, one overtriggers.

**Emitted set is a strict subset of what the scanner marks.** Only the two I28 families have crosswalk rows; the other eight are counted as declined with `reason=no-severity-crosswalk-row`. "No crosswalk row, no relay" made visible rather than enforced by silence.

**The read-only contract is unchanged.** Persisting is opt-in behind `--persist-findings`; a bare invocation reports and stops. The file is a proposal artifact for a human-gated relay, never an applied edit.

**One contract refinement the end-to-end run forced.** #3120 asked for a test proving the directive survives "byte-for-byte". Run against a real leading wrapper, that fails: dropping `CRITICAL: You MUST ` promotes the next word to sentence-initial position, so `…MUST resolve the item id` → `Resolve the item id` and exactly one byte changes. It is forced by the edit, not chosen — and the official source's own worked example makes the same change (`use` → `Use`), so a strict byte-equality contract would reject upstream's own demonstration. The contract now asserts verbatim survival **apart from that forced capitalization**; any other wording change means the remediation overreached.

**One out-of-theme change, called out deliberately.** SKILL.md had 4 lines of headroom under the 500-line hard cap, so documenting a new capability required moving something out. The Phase D state-key mechanics went to `context/report-keying.md` as a progressive-disclosure spoke. That block also restated rules the `plugin-data-report-keying` convention already owns, so the spoke points at the owner instead. Every removed claim was verified recovered.

## Review rounds

**Nine automated review passes ran across three reviewers** — Codex, the security-review lane (six passes), and the code-review lane (three). Six findings total, all reproduced empirically before fixing rather than accepted on assertion, and all fixed:

1. **Body-scope fence bypass on non-canonical `---` delimiters (P1).** Exact `"---"` equality is stricter than `skill_frontmatter::extract`'s `^---[[:space:]]*$`. A CRLF or trailing-whitespace delimiter is real frontmatter to the gate but was invisible to the fence, so the block read as body. Measured: a CRLF fixture emitted a finding pointing at its own `description` line. Because `descr()` shares the detection, the quoted-trigger fence was disabled in the same case, and because the writer recomputed the *same* predicate, both layers failed together rather than independently — the one property the defense-in-depth design was meant to buy. All three checks now use the authoritative pattern.
2. **Out-of-repo `Location` reached the relay (P1).** Declined and counted, per above.
3. **Silent carve-out declines (P2).** `persist-findings.md` told the model to drop carve-out rows before the writer saw them while the same file requires every decline counted. New `--declined-carveout N` flag records the count.
4. **YAML-indicator branch names misparsed (P2).** `@foo`/`!foo`/`#foo` are valid git branches; emitted plain, `#foo` reads as a comment. Quoting is **conditional** — an ordinary branch stays a byte-identical plain scalar, so the common path and parity with the sibling `ai-slop` producer are preserved. (That producer has the same unquoted interpolation; flagged as a fleet-wide edge case, left out of scope here.)
5. **`source-line-unreadable` decline had zero test coverage** — the only decline path without an assertion. Covered by case 8b, verified load-bearing by mutation (deleting the branch fails exactly its three assertions).
6. **Case-ordering nit** — case 8b sat out of reading order; moved to sit with case 8.

The final pass of each lane skip-gated, having nothing new to review. The last substantive security verdict: **"no outstanding security issues."**

`fixture-git-isolation-gate` separately caught that the new tests build a `git init` fixture without clearing the ambient git environment — a real hazard, since an inherited absolute `GIT_DIR` outranks `git -C` and would write the fixture identity into the caller's `.git/config`. Cleared at the top of the suite, matching plugin-suite precedent; not added to the baseline file, which records pre-existing debt only.

## Verification

Run, not inspected.

**End-to-end proof** (acceptance criterion 3) against a sample surface carrying coercive emphasis in *both* frontmatter and body:

1. Detector runs body-scoped → 2 candidates (frontmatter correctly excluded)
2. `emit-findings.sh` writes to the rung-5 findings home for this branch
3. Every clause of `fix-pass-mode.md` "Step 1" verified against the written file: `.md` extension, `type: review-findings`, `branch:` exact match, parseable table, no subtracting `fix-pass-record` — all PASS
4. Remediations applied at each `Location`: frontmatter digest unchanged, `description` and `when_to_use` byte-identical, quoted trigger phrases intact, directive verbatim modulo the one forced capital, instruction not deleted, protected-content line untouched
5. Re-scan returns no candidates — no stale finding survives its own remediation

**Gates** (final):

| Check | Result |
|---|---|
| `emit-findings.test.sh` | All 75 checks passed |
| `instruction-scan.test.sh` | All 92 checks passed |
| `conflict-scan.test.sh` | All 48 checks passed |
| `skill-quality:check` (source checkout) | **PASS** — 0 errors; 483/500 lines; all 13 base-ref trigger phrases preserved |
| `check-fixture-git-isolation.sh` | OK (87 isolated, 0 baselined) |
| shellcheck / shfmt / markdownlint / typos / editorconfig | clean |

Every suite was additionally verified **under CI's detached-HEAD condition**, which is what surfaced the `--branch` resolution ordering that made one assertion measure the wrong thing.

The negative test feeds the writer *deliberately unfenced* scan output and asserts no emitted row points at a description or `when_to_use`. Protected content gets one no-flag line per category the container spec names. Fence regressions cover CRLF, trailing space, trailing tab, and unclosed frontmatter in both layers. The remaining `check-skill` warning (`criteria.md:1020`) reproduces on the base ref.

**Pre-PR diff gate:** frontmatter blast radius is nil — only `argument-hint` changed, `description` byte-identical. The cross-plugin contract change is purely additive; no existing crosswalk row or rule modified.

## Related

Refs #3118 — parent spec container; this is its tracer-bullet slice, carrying the producer wiring D1 and D3 will reuse.

Two container-level observations, neither blocking this slice:

- #3118 names `docs/specs/agent-doc-register-detectors-brief.md` as its authoring surface; that file does not exist in the repo.
- I28's V1 selection is deliberately narrower than its Detect prose — whole bolded sentences and general all-caps imperative runs are judged by the model lane but not mechanically selected, both being too common in ordinary technical prose to select without swamping the relay. Recorded in `criteria.md` rather than left as a silent gap, following the precedent where `ai-slop` deferred its third negative-parallelism pattern.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01Xn6aUXRQSYx4cGuP5cDxbF

### CODE REVIEW

File: plugins/claude-config/skills/audit-instructions/scripts/emit-findings.sh [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+  # Frontmatter close line for <file>, or 0 when there is none. Recomputed here
+  # rather than trusted from the caller (header comment: BODY-SCOPE FENCE).
+  # An unclosed leading `---` fences the whole file, the fail-safe direction.
+  function fm_end(file,   line, n, fmclose, result) {
+    if (file in fmcache) return fmcache[file]
+    n = 0; fmclose = -1
+    while ((getline line < file) > 0) {
+      n++
+      if (n == 1) { if (line != "---") { fmclose = 0; break } ; continue }
+      if (line == "---") { fmclose = n; break }
```
[the remark this answers is not in the document]
Comment (kyle-sexton) [reply plugins/claude-config/skills/audit-instructions/scripts/emit-findings.sh]: Confirmed and fixed — this was the most important finding on the PR, and the framing was more accurate than my own first fix.

I initially closed only the CRLF half by stripping a terminal CR. That was insufficient: the real defect is that exact `"---"` equality is **stricter than the parser the fence exists to agree with**, and the mismatch runs in the dangerous direction. All three checks now use `skill_frontmatter::extract`'s `^---[[:space:]]*$`, so the three readers agree rather than the fence being only as strong as the loosest one. `[[:space:]]` subsumes CR, so it replaces the strip on delimiters; the excerpt keeps its own so no stray CR reaches a table cell.

Reproduced before fixing — a CRLF fixture emitted a finding pointing at its own `description` line:

```
| 1 | IMPORTANT | high | crlf.md:2 | ... rule-coercive-emphasis marker="CRITICAL:" ...
```

Your note that `descr()` shares the detection was correct and I'd have missed it: the quoted-trigger fence was disabled in the same case. And the observation that both layers fail on identical input rather than independently is the sharpest point here — that's precisely the property the defense-in-depth design was meant to buy, and it wasn't buying it.

Regression cases added in **both** layers for CRLF, trailing space, trailing tab, and the unclosed-frontmatter variant. 72 producer + 92 scanner checks green.

Fixed in bab7be71 (and 794851a1 for the CRLF half).

---
_Generated by [Claude Code](https://claude.ai/code)_

File: plugins/claude-config/skills/audit-instructions/scripts/emit-findings.sh [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+
+    if (lno + 0 <= fm_end(file)) { declined_frontmatter[id]++; next }
+
+    text = source_line(file, lno + 0)
+    if (text == "") { declined_unreadable[id]++; next }
+
+    if (quotes_trigger(file, text)) { declined_trigger[id]++; next }
+
+    loc = file
+    if (repo_root != "" && index(loc, repo_root "/") == 1) loc = substr(loc, length(repo_root) + 2)
```
[the remark this answers is not in the document]
Comment (kyle-sexton) [reply plugins/claude-config/skills/audit-instructions/scripts/emit-findings.sh]: Confirmed and fixed in 794851a1.

Reproduced first — a hit in a user-level `CLAUDE.md` outside the repo emitted an absolute `Location`:

```
| 1 | IMPORTANT | high | /tmp/.../fakehome/CLAUDE.md:3 | ...
```

`emit-findings.sh` now declines any row whose path is not under the repo root, counted as `reason=outside-repo-root`. I took your "remain in the human report" option rather than inventing a new disposition: those findings are still real, they just can't be relay-bound, and `persist-findings.md` now says so and routes upstream-owned surfaces to the owning repository per the skill body's existing routing rule.

Worth noting this fence also caught two of my own test fixtures that lived in `/tmp` — they had to move inside a throwaway git repo to keep measuring what they were written to measure. Good sign the fence is load-bearing.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: plugins/claude-config/skills/audit-instructions/scripts/emit-findings.sh [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+    if (length(excerpt) > 160) excerpt = substr(excerpt, 1, 157) "..."
+
+    rows[++nemit] = "| " rule_tier(id) " | high | " loc ":" lno " | claude-config:audit-instructions | " \
+      esc(rid " " fired_marker(id, text) " -- " excerpt) " | " esc(rule_action(id)) " |"
+    seen[id]++
+    next
+  }
+
+  END {
+    printf "---\ntype: review-findings\ndate: %s\nbranch: %s\n---\n\n", date_utc, branch
```
[the remark this answers is not in the document]
Comment (kyle-sexton) [reply plugins/claude-config/skills/audit-instructions/scripts/emit-findings.sh]: Confirmed and fixed in 794851a1 — with one deliberate deviation from the suggestion.

Verified all three are valid git branch names (`git check-ref-format --branch`), and that each emitted a plain scalar that misparses.

You suggested serializing as a quoted scalar. I made the quoting **conditional** rather than unconditional: only when the value would actually misparse (leading YAML indicator, `: `, ` #`, empty, or trailing whitespace). Reason — the consumer admits a candidate only on an exact `branch:` match, and the sibling `ai-slop` producer emits plain scalars today. Unconditional quoting would change the wire format for every ordinary branch at once, which risks breaking the common path to fix an edge case. Conditional quoting leaves `main` and `feat/3120-thing` byte-identical while closing the misparse:

```
branch '@foo'          -> branch: "@foo"
branch '#foo'          -> branch: "#foo"
branch 'feat/3120-thing' -> branch: feat/3120-thing
```

Covered by tests over `@foo !foo #foo *foo &foo` (quoted) and `main`, `feat/3120-thing`, `release-1.2_x` (unquoted).

Flagging for the maintainer: `ai-slop`'s `emit-findings.sh` has the same unquoted interpolation, so this is a fleet-wide edge case rather than one local to this PR. I've left that alone as out of scope here.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: plugins/claude-config/skills/audit-instructions/context/persist-findings.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+| Scanner family | Rule id | Tier |
+|---|---|---|
+| `I28-a` | `claude-config/audit-instructions/rule-coercive-emphasis` | IMPORTANT |
+| `I28-b` | `claude-config/audit-instructions/rule-blanket-tool-default` | IMPORTANT |
+
+The model lane's criteria carve-outs still apply **before** persistence: emphasis guarding a
+destructive, security, or permission gate, a stated hard precondition, and a document *about* the
+pattern are not findings (reference/criteria.md, I28). The scanner over-produces by design; drop
+those candidates from the scan output handed to `--from` rather than emitting and retracting.
```
[the remark this answers is not in the document]
Comment (kyle-sexton) [reply plugins/claude-config/skills/audit-instructions/context/persist-findings.md:76]: Confirmed and fixed in 794851a1. This was a genuine internal contradiction in the document — it told the model to drop carve-out rows before the writer saw them, while the same file requires every decline to be counted.

Took your suggestion directly: `emit-findings.sh` gains `--declined-carveout <n>`, which records the count as its own `## Surfaces` line, and `persist-findings.md` now instructs passing it (omit when zero). A non-numeric value exits 2 rather than silently emitting a bogus count.

```
Declined candidates: I28 count=3 reason=criteria-carve-out (model lane; see reference/criteria.md I28)
```

I kept the count rather than a full record: the carve-out decision is the model lane's and its reasoning belongs in the human report, so what `## Surfaces` needs is honest arithmetic about how many candidates were examined — not a second copy of the rationale in a file the relay consumes.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: plugins/claude-config/skills/audit-instructions/scripts/instruction-scan.sh [hunk 5]
```diff
@@ -158,6 +181,30 @@ I25_ERE="${WB_L}temperature${WB_R}|${WB_L}top_p${WB_R}|${WB_L}top_k${WB_R}"
...
+# Print the line number of a leading YAML frontmatter block's CLOSING `---`, or 0
+# when the file has none. Frontmatter is recognized only when `---` is the very
+# first line, which is the form every skill, agent, and output-style surface
+# uses; a `---` thematic break mid-document therefore opens nothing. An UNCLOSED
+# leading `---` returns the file's line count, fencing the whole file rather than
+# none of it — the fail-safe direction, since the alternative would route a
+# malformed-frontmatter description straight to an apply relay.
+frontmatter_end() {
+  local file="$1"
+  [[ "$(head -n 1 "$file" 2>/dev/null)" == "---" ]] || {
```
[the remark this answers is not in the document]
Comment (kyle-sexton) [reply plugins/claude-config/skills/audit-instructions/scripts/instruction-scan.sh]: Confirmed and fixed in bab7be71 — `frontmatter_end` now uses `^---[[:space:]]*$` in both its opening and closing checks, matching `skill_frontmatter::extract`.

Your point that the two layers fail on *identical* input rather than independently is the one that changed how I fixed this. My first pass (794851a1) stripped a terminal CR in both layers, which closed the CRLF case but left them still sharing the same too-strict predicate — so trailing whitespace would have taken both out together exactly as you describe. Aligning all three on the authoritative pattern is the actual fix.

Scanner-side regression cases added for CRLF, trailing space, trailing tab, and unclosed frontmatter (92 checks green). Full detail on the shared thread: #discussion_r3837763923.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: plugins/claude-config/skills/audit-instructions/scripts/emit-findings.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (kyle-sexton) [reply plugins/claude-config/skills/audit-instructions/scripts/emit-findings.sh]: Duplicate of #discussion_r3837748428 / #discussion_r3837763923 — same finding, answered in full there. Fixed in bab7be71: all three delimiter checks now use `^---[[:space:]]*$`, matching `skill_frontmatter::extract`, with CRLF and trailing-whitespace regression cases in both layers.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: plugins/claude-config/skills/audit-instructions/scripts/emit-findings.sh [hunk 6]
```diff
@@ -0,0 +1,348 @@
...
+    lno = rest; sub(/^.*:/, "", lno)
+    file = rest; sub(/:[^:]*$/, "", file)
+
+    rid = rule_id(id)
+    if (rid == "") { declined_nocrosswalk[id]++; next }
+
+    if (lno + 0 <= fm_end(file)) { declined_frontmatter[id]++; next }
+
+    text = source_line(file, lno + 0)
+    if (text == "") { declined_unreadable[id]++; next }
```
[the remark this answers is not in the document]
Comment (kyle-sexton) [reply plugins/claude-config/skills/audit-instructions/scripts/emit-findings.sh:288]: Confirmed and fixed in c14c5988. Verified the gap before acting rather than taking it on assertion:

```
no-severity-crosswalk-row    test assertions: 1
frontmatter                  test assertions: 16
quoted-trigger-phrase        test assertions: 2
outside-repo-root            test assertions: 1
criteria-carve-out           test assertions: 2
source-line-unreadable       test assertions: 0   ← the only uncovered path
```

New case 8b feeds a row pointing past EOF and asserts three things: no findings row, the decline counted as `reason=source-line-unreadable`, and no row bearing the phantom location.

Your reachability argument is what made this worth doing now rather than filing — the scan output and the source file are read at different moments, and `persist-findings.md` puts a model edit between them, so this is not a synthetic-only branch.

**I also checked the new case is load-bearing** rather than passing vacuously: deleting `if (text == "") { declined_unreadable[id]++; next }` fails exactly those three assertions and nothing else.

One thing worth recording since it nearly shipped: written against the shared `OUT` variable, the case clobbered what cases 10 and 11 read, silently breaking three unrelated assertions. Running the whole suite rather than only the new greps caught it; it now uses its own variable, with an inline note so the next case added here does not repeat it.

75 producer checks (was 72); scanner 92 and conflict 48 unchanged. `check-skill` PASS, `fixture-git-isolation` OK.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: plugins/claude-config/skills/audit-instructions/scripts/emit-findings.sh [hunk 7]
```diff
@@ -0,0 +1,286 @@
...
+      n++
+      if (n == 1) { if (line != "---") { fmclose = 0; break } ; continue }
+      if (line == "---") { fmclose = n; break }
+    }
+    while ((getline line < file) > 0) n++
+    result = (fmclose == -1) ? n : fmclose
+    close(file)
+    fmcache[file] = result
+    return result
+  }
```
[the remark this answers is not in the document]
Comment (kyle-sexton) [reply plugins/claude-config/skills/audit-instructions/scripts/emit-findings.sh:214]: Confirmed and fixed — your static trace was exactly right, and I reproduced it before fixing: a CRLF fixture emitted a finding pointing at its own `description` line.

Your suggested fix (strip a trailing `\r` before each comparison) was what I did first, in 794851a1. A parallel finding then showed it was insufficient: the deeper defect is that exact `"---"` equality is **stricter than `skill_frontmatter::extract`'s `^---[[:space:]]*$`**, the pattern `check-skill.sh` actually parses frontmatter with — so trailing whitespace bypassed the fence too, and both layers still shared one too-strict predicate. bab7be71 aligns all three checks on the authoritative pattern; `[[:space:]]` subsumes CR, so it replaces the strip on delimiters. The excerpt keeps its own strip so no stray CR reaches a table cell.

Your point that "there is no surviving layer on a CRLF input" is the one that mattered most — that's precisely the property the defense-in-depth design was supposed to buy, and it wasn't buying it because the writer recomputed the *same* predicate rather than an independent one.

Regression cases in both layers for CRLF, trailing space, trailing tab, and unclosed frontmatter. Full detail: #discussion_r3837763923.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: plugins/claude-config/skills/audit-instructions/scripts/emit-findings.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (kyle-sexton) [reply plugins/claude-config/skills/audit-instructions/scripts/emit-findings.sh]: Confirmed and fixed in 794851a1. Reproduced first — a hit in a user-level `CLAUDE.md` outside the repo emitted an absolute `Location`:

```
| 1 | IMPORTANT | high | /tmp/.../fakehome/CLAUDE.md:3 | ...
```

I took the second of your two options — decline here and count it — rather than restricting the `--from` set upstream. Reason: the restriction would live in the model lane, where nothing enforces it, while the writer-side fence holds regardless of who assembles the input. That is the same argument the body-scope fence rests on, so applying it inconsistently here would have been the odd choice.

Declined rows are counted as `reason=outside-repo-root`, matching the existing decline pattern you pointed at. `persist-findings.md` routes them to the human report, and to the owning repository where the surface is upstream-owned.

Your framing of it as a trust-boundary contract violation rather than an exploitable bug is right, and it is why this got a fence rather than a filed follow-up.

Two of my own test fixtures living in `/tmp` were caught by the new fence and had to move inside a throwaway git repo — a decent sign it is load-bearing.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: plugins/claude-config/skills/audit-instructions/scripts/emit-findings.test.sh [hunk 8]
```diff
@@ -0,0 +1,411 @@
...
+  fail "the directive survives verbatim apart from forced capitalization" \
+    "expected '$ORIG_DIRECTIVE', got '$(printf '%s' "$REMEDIATED" | tr '[:upper:]' '[:lower:]')'"
+fi
+# And the remediated line must no longer be a candidate: no stale finding survives
+# its own remediation.
+printf '%s\n' "$REMEDIATED" >"$TEST_TMPDIR/remediated.md"
+POST=$(bash "$SCAN" --body-only "$TEST_TMPDIR/remediated.md")
+assert_contains "the remediated line is no longer a candidate" "$POST" "No instruction candidates found."
+
+# --- Case 8b: a row pointing past EOF is declined, not emitted blank ---------
```
[the remark this answers is not in the document]
Comment (kyle-sexton) [reply plugins/claude-config/skills/audit-instructions/scripts/emit-findings.test.sh]: Fair nit — fixed in bba7e8c3, though not by either route exactly as offered.

`9g` would have been wrong for where the block actually sat. Checked the real order before acting:

```
177:  Case 8
246:  Case 8b   ← between 9b and 9c, not after 9f
266:  Case 9c
342:  Case 9f
```

`9g` implies it follows `9f`; it preceded `9c`. That rename would have traded one ordering lie for another.

Took your second option instead — moved the block to sit right after Case 8. Better here on both counts: it's where the existing `8b` label already claims it belongs, so nothing needs renumbering, and it's the right neighbour thematically (Case 8 decides which rows are **admitted**, 8b decides which are **declined**; the fence cases at 9c–9f are a different concern).

Pure block move, no assertion touched. Case order is now strictly ascending end to end. Re-verified after moving: 75 producer checks pass, and the mutation check still catches exactly the same three assertions when the `declined_unreadable` branch is deleted — so the case is still load-bearing where it now sits, not merely still green.

Thanks for independently confirming the `EOF_OUT`/`EOFROW` isolation — that's the part I got wrong on the first attempt at this case, and it's worth having a second pair of eyes on.

---
_Generated by [Claude Code](https://claude.ai/code)_
